Architecture.md

```diff
@@ -4,79 +4,150 @@
+
+- NestJS-express is nearly on par with Express for the average response time (`65.44 ms` versus `61.88 ms`)
+- NestJS-express is twice as fast in maximum response time (only `325 ms` against Express' `747 ms`)
+- NestJS-express can handle nearly twice as many requests concurrently, for the minimum amount doable (`14183 req/s` versus `8407 req/s`)
+- NestJS-express is on par with Express for the number of requests handled concurrently, for the average amount doable (`15640` versus `16454.41`)
+
+The API is currently running on NestJS-express v10.0.3, which just released. Two years have passed since the last benchmarks, and we can expect NestJS to be even faster as of now.
+
+##### Authentication strategies
+
+Regarding the authentication strategy, our team decided to create a `GET /auth/login` endpoint which returns a both an JWT HttpOnly cookie as well as a Bearer token. The cookies keep the user safe from XSS (Cross-Site Scripting) while the Bearer token allows users to log in via the CLI.
+
+##### Mail
+
+With sending and receiving emails being an important part of this application for the user, we needed a local way to test these functions without actually needing to connect it to a private email every time. After some search we came across [mailhog](https://github.com/mailhog/MailHog). Mailhog allows anyone to create a temporary local smtp server and send and receive emails through it. Even though you are not able to send emails to real email adresses, you can send and receive local emails, which allowed us to test our code.
+
+##### GRPC
+
+To communicate with the other services, especially pomegranate and the git-repo-manager, we used [grpc](https://grpc.io/), due to its fast performance and low latency. It also allows us to easily create nest clients for the other services, which we can use to communicate with them. Another great advantages of grpc is that it is language agnostic, which means that we can use it to communicate with services written in other languages. Creating the proto files allowed us to define contracts between the services, which made it easier to develop the services independently.
+
+##### SSH key
+
+To be able to push their projects onto the Git server, each user needs to associate at least one ssh key with their account. Using the command line, they will then be able to push their repositories to the server.
+
+##### Administrators
+In addition to the permissions of a normal users, administrators have a few extra perks. They are able to view the SSH keys of every Client as well as their personal non-critical information (email, username)- and even delete a Client.
+At the moment, it is not possible to get an administrator, as they don't have the right to modify other Client's information.
```

Comment:
I'm not sure what "get an administrator" means. Maybe it's about not being able to become an administrator or rather make another user an administrator?

Reply:
both
- being able to become an administrator
- update another user to make them an administrator

I'll try to reformulate it